Atom2. app_bench. ошибка типизации

web.js:1 mol/app/bench/bench/-view.tree/bench.view.tree.ts(188,8): error TS2339: Property 'col_head_label' does not exist on type '$mol_bench'.

http://127.0.0.1:8080/mol/app/bench/index.html

#320

--- a/app/bench/bench/bench.view.tree
+++ b/app/bench/bench/bench.view.tree
@@ -20,8 +20,8 @@ $mol_app_bench $mol_view
 							<= param_fields /
 					<= Result $mol_bench
 						result <= result null
-						col_head_label!id /
-							<= result_col_title!id /
+						col_head_title!id
+							<= result_col_title!id \
 						col_sort?val <=> result_col_sort?val \
 		<= Sandbox_page $mol_page
 			title <= sandbox_title @ \Sandbox

--- a/app/bench/bench/bench.view.ts
+++ b/app/bench/bench/bench.view.ts
@@ -139,8 +139,8 @@ namespace $.$$ {
 		}
 
 		result_col_title( col_id : string ) {
-			if( col_id === 'sample' ) return [ this.result_col_title_sample() ]
-			return this.step_title( col_id )
+			if( col_id === 'sample' ) return this.result_col_title_sample()
+			return this.step_title( col_id ).join(' ')
 		}
 		
 		step_title( step : string ) {

--- a/app/lamps/lamps.view.tree
+++ b/app/lamps/lamps.view.tree
@@ -7,13 +7,12 @@ $mol_app_lamps $mol_book
 			body_scroll_top?val <=> menu_scroll_top?val 0
 			body /
 				<= Menu $mol_list
-					rows /
+					rows <= lamp_rows /
 						<= Filter_panel $mol_float
 							sub /
 								<= Filter $mol_code
 									hint <= filter_hint @ \Filter...
 									value?val <=> filter?val \
-						<= lamp_rows /
 		<= Main_page $mol_page
 			minimal_width 400
 			title <= title \

--- a/app/lamps/lamps.view.ts
+++ b/app/lamps/lamps.view.ts
@@ -33,7 +33,7 @@ namespace $.$$ {
 		}
 		
 		lamp_rows() {
-			return super.lamp_rows().concat(this.lamps().map( lamp => this.Lamp_row( lamp[ 'no' ] ) ))
+			return [...super.lamp_rows(), ...this.lamps().map( lamp => this.Lamp_row( lamp[ 'no' ] ) )]
 		}
 		
 		lamp_title( id : string ) {

--- a/app/lamps/lamps.view.ts
+++ b/app/lamps/lamps.view.ts
@@ -33,7 +33,7 @@ namespace $.$$ {
 		}
 		
 		lamp_rows() {
-			return this.lamps().map( lamp => this.Lamp_row( lamp[ 'no' ] ) )
+			return super.lamp_rows().concat(this.lamps().map( lamp => this.Lamp_row( lamp[ 'no' ] ) ))
 		}
 		
 		lamp_title( id : string ) {

--- a/app/report/report.view.tree
+++ b/app/report/report.view.tree
@@ -4,10 +4,9 @@ $mol_app_report $mol_page
 		<= descriptor $mol_view
 			sub / <= description \
 		<= tabler $mol_app_report_tabler
-			rows /
+			rows <= rows /
 				<= headRower $mol_app_report_rower
 					cells <= headCells /
-				<= rows /
 	rower!id $mol_app_report_rower
 		cells <= rowerCells!id /
 	cell!id $mol_app_report_cell

--- a/app/report/report.view.ts
+++ b/app/report/report.view.ts
@@ -188,7 +188,7 @@ namespace $.$$ {
 		}
 		
 		rows() {
-			const rows : $mol_app_report_rower[] = []
+			const rows : $mol_app_report_rower[] = super.rows().slice()
 			
 			const visit = ( pos : number[] , format : $mol_app_report_formatRow )=> {
 				rows.push( this.rower( pos ) )

